src/Dhall.hs

```diff
@@ -90,7 +90,6 @@ import Control.Exception (Exception)
 import Control.Monad.Trans.State.Strict
 import Data.Functor.Contravariant (Contravariant(..), (>$<))
 import Data.Functor.Contravariant.Divisible (Divisible(..), divided)
-import Data.Monoid ((<>))
```

Comment:
`stack build --stack-yaml stack-lts6.yaml` fails with
```console
  C:\dhall-haskell\src\Dhall.hs:153:16:
  Not in scope: ‘<>’ 
    Perhaps you meant one of these:
      data constructor ‘Data.Sequence.:>’ (imported from Data.Sequence),
     ‘<|>’ (imported from Control.Applicative),
     ‘<*>’ (imported from Prelude)
```
Adding it again makes it work and the builds with `stack.yaml` and `stack-lts11.yaml` (`ghc-8.2.2`) work too.

Reply:
It seems in a previous version the module imported `Data.Semigroup`, so it was necessary to remove `Dhall.Monoid.(<>)`, but without any of them `<>` is out of scope for previous versions of ghc

Reply:
I tried searching stackage.org and it seems `Data.Semigroup` wasn't available in `base` at the time. Is there a reason we're supporting and old lts? If it's to support stack + a certain version of ghc we can just specify which compiler to use in the yaml file instead of the lts.

Reply:
i am afraid that (etlas)[https://github.com/typelead/etlas], that uses `dhall` through (dhall-to-etlas)[https://github.com/eta-lang/dhall-to-etlas] is being compiled with lts-6 and afaik it can't be compiled with newer lts's

Reply:
But adding `import Data.Monoid ((<>))` seems to fix it, at least for `lts-6`, `lts-11` and `lts-12`, could it cause some problem now or in the future?

Reply:
Woof! Is there any plan to update that? I don't think it would change things in the future. I think we just need to import `Data.Semigroup (Semigroup)` for the class, and then we can import `Data.Monoid ((<>))` for the operator

Reply:
Maybe @rahulmutt (owner of etlas) could answer it better than me :wink:
The main goal is [make dhall the default configuration format for etlas](https://github.com/typelead/etlas/issues/66)

Reply:
I've currently got a `stack build` going with `etlas` and this [yaml](https://github.com/FintanH/etlas/blob/fintan/update-lts/stack-ghc822.yaml)

Reply:
Ah ok. I'll move the discussion to those repos. In the meantime, I did a fix up of this branch for both stack yamls. I'll make a PR against this branch now

Reply:
https://github.com/dhall-lang/dhall-haskell/pull/616